src/bin/main.rs

```diff
@@ -0,0 +1,127 @@
+extern crate observer;
+use chrono::Utc;
+use core::borrow::{Borrow, BorrowMut};
+use observer::context::{Context, Frame};
+use observer::event::Event;
+use observer::observer::{observe, save};
+use observer::AResult;
+use serde::ser::Serialize;
+use serde_derive::{Deserialize, Serialize};
+use std::collections::LinkedList;
+use std::thread;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct UserId {
+    id: i32,
+}
+
+impl Event<UserId> for UserId {
+    fn map(&self, ctx: &RefCell<Context>, _data: UserId) -> AResult<UserId> {
```

Comment:
Not fond of the RefCell here. Put RefCell inside the context struct instead.

Reply:
Like we have done for `In` struct.

Reply:
Got it